acr-credential-provider.spec

```diff
@@ -0,0 +1,155 @@
+# Built from this template:
+# https://github.com/openshift/release/blob/master/tools/hack/golang/package.spec
+#
+# See also: https://github.com/openshift/kubernetes/blob/master/openshift.spec
+
+#debuginfo not supported with Go
+%global debug_package %{nil}
+
+# modifying the Go binaries breaks the DWARF debugging
+%global __os_install_post %{_rpmconfigdir}/brp-compress
+
+# %commit and os_git_vars are intended to be set by tito custom builders
+# provided in the .tito/lib directory. The values in this spec file will not be
+# kept up to date.
+%{!?commit: %global commit HEAD }
+%global shortcommit %(c=%{commit}; echo ${c:0:7})
+# os_git_vars needed to run hack scripts during rpm builds
+%{!?os_git_vars: %global os_git_vars OS_GIT_VERSION='' OS_GIT_COMMIT='' OS_GIT_MAJOR='' OS_GIT_MINOR='' OS_GIT_TREE_STATE='' }
+
+%if 0%{?skip_build}
+%global do_build 0
+%else
+%global do_build 1
+%endif
+%if 0%{?skip_prep}
+%global do_prep 0
+%else
+%global do_prep 1
+%endif
+
+%if 0%{?fedora} || 0%{?epel}
+%global need_redistributable_set 0
+%else
+# Due to library availability, redistributable builds only work on x86_64
+%ifarch x86_64
+%global need_redistributable_set 1
+%else
+%global need_redistributable_set 0
+%endif
+%endif
+%{!?make_redistributable: %global make_redistributable %{need_redistributable_set}}
+
+#
+# Customize from here.
+#
+
+%global golang_version 1.21.0
+%{!?version: %global version 0.0.1}
+%{!?release: %global release 1}
+%global package_name acr-credential-provider
+%global product_name acr-credential-provider
+
+Name:           %{package_name}
+Version:        %{version}
+Release:        %{release}%{?dist}
+Summary:        Azure acr kubelet image credential provider
+License:        ASL 2.0
+
+Source0:        %{name}.tar.gz
+BuildRequires:  bsdtar
+BuildRequires:  golang >= %{golang_version}
+Provides:       ose-azure-acr-image-credential-provider = %{version}-%{release}
+
+# If go_arches not defined fall through to implicit golang archs
+%if 0%{?go_arches:1}
+ExclusiveArch:  %{go_arches}
+%else
+ExclusiveArch:  x86_64 aarch64 ppc64le s390x
+%endif
+
+### AUTO-BUNDLED-GEN-ENTRY-POINT
+
+%description
+This package provides the acr-credential-provider. It is a kubelet image
+credential provider. It dynamically provides kubelet with credentials to pull
+images from acr. For more information see:
+https://kubernetes.io/docs/tasks/administer-cluster/kubelet-credential-provider/
+
+%prep
+%if 0%{do_prep}
+%setup -q
+%endif
+
+%build
+
+%if 0%{do_build}
+# Expand os_git_vars macro so we can use the variables it defines
+%{os_git_vars}
+
+%global version_pkg sigs.k8s.io/cloud-provider-azure/pkg/version
+%global client_go_version_pkg k8s.io/client-go/pkg/version
+%global build_date %(date -u +'%Y-%m-%dT%H:%M:%SZ')
+
+LDFLAGS="-s -w"
```

Comment:
We set some `ldflags` (not as extensively as this) in the `openshift-hack/build-go.sh` script. 

```bash
VERSION_PKG=sigs.k8s.io/cloud-provider-azure/pkg/version

GLDFLAGS="-extldflags '-static'"
GLDFLAGS="$GLDFLAGS -X $VERSION_PKG.gitVersion=${REPO}"
GLDFLAGS="$GLDFLAGS -X $VERSION_PKG.gitCommit=${VERSION_OVERRIDE}"
GLDFLAGS="$GLDFLAGS -X $VERSION_PKG.buildDate=$(date -u +'%Y-%m-%dT%H:%M:%SZ')"
```

Are these sufficient? We could always override `GLDFLAGS` when building the spec.

Reply:
It is probably sufficient, as it's what we are doing in `cloud-provider-aws`

Reply:
It's actually not reporting a version when starting the credential provider. 

Only the CCM and CNM report a version: 

https://github.com/openshift/cloud-provider-azure/blob/07508efb67bd927943b7613d7bfba550301f2663/cmd/cloud-controller-manager/app/controllermanager.go#L302-L304

https://github.com/openshift/cloud-provider-azure/blob/07508efb67bd927943b7613d7bfba550301f2663/cmd/cloud-node-manager/app/nodemanager.go#L101-L103

Reply:
I think we're fine to not report a version here, through logs or a flag. The RPM we ship will be versioned.

The ecr one doesnt log anything, but you can pass `-v`. 

The gcr one doesnt log anything, and you can't pass a flag either.